microdnf/context.cpp

```diff
@@ -630,6 +652,19 @@ void Context::download_and_run(libdnf::base::Transaction & transaction) {
     // TODO(mblaha): print a summary of successfull transaction
 }
 
+bool parse_add_spec(const std::string & spec, std::set<std::string> & pkg_specs, std::set<std::string> & filepaths) {
+    bool inserted;
+    const std::string_view ext(".rpm");
+    if (spec.length() > ext.length() && spec.compare(spec.length() - ext.length(), ext.length(), ext) == 0) {
+        auto ret = filepaths.emplace(spec);
+        inserted = ret.second;
```

Comment:
The original code in DNF also handled arguments with schemes as filenames. What your suggestion about it?
```
        if value in tmp_set:
            continue
        tmp_set.add(value)
        schemes = dnf.pycomp.urlparse.urlparse(value)[0]
        if value.endswith('.rpm'):
            namespace.filenames.append(value)
        elif schemes and schemes in ('http', 'ftp', 'file', 'https'):
            namespace.filenames.append(value)
        elif value.startswith('@'):
            namespace.grp_specs.append(value[1:])
        else:
            namespace.pkg_specs.append(value)```

Reply:
Microdnf does not yet support command line packages from remote files. So schema support doesn't make sense.
When I write support for remote files, I will have to do schema parsing support.